### PR TITLE
Stringsifter docker

### PR DESCRIPTION
**Pull request by omerKarkKatz**
## Status
Ready/In Progress/In Hold (Reason for hold)

## Related Content Pull Request
Related PR: link to the PR at demisto/content

## Related Issues
Related: https://jira-hq.paloaltonetworks.local/browse/CIAC-3263

## Description
This is a new docker for the string sifter automation. notice that the lightgbm module is restricted to 3.3.0 for bugs mentioned by Ben melamed

### CODE REVIEW

**Comment by GuyAfik** [comment]
Please also add `verify.py` if possible

**Comment by efelmandar** [comment]
.

File: docker/stringsifter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM demisto/python3-deb:3.8.6.14507
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply docker/stringsifter/Dockerfile:1]: Seems like @wissamg tried using python3.10 and `FROM demisto/python3-deb:3.10.5.31873` with no success. Only after moving it to  `FROM demisto/python3-deb:3.8.6.14507` it worked.

File: docker/stringsifter/Pipfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+lightgbm = "==3.3.0"
+numpy = "~= 1.17"
+scikit-learn = "== 0.23.2"
+joblib = ">=0.13.2,<0.17"
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply docker/stringsifter/Pipfile:10]: The example docker image from [StringSifter](https://github.com/mandiant/stringsifter/blob/master/Pipfile) repo was taken. It was locked there originally.

File: docker/stringsifter/Pipfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+stringsifter= "~= 2.0"
+
+[dev-packages]
+pytest = "*"
+setuptools = "*"
+twine = "*"
+wheel = "*"
+
+[requires]
+python_version = "3.8"
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply docker/stringsifter/Pipfile:21]: I guess it should be consistent with the python 3.8 used in the dockerfile. Again originally @wissamg used python 3.10 and moved it back to python 3.8.

File: docker/stringsifter/Pipfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply docker/stringsifter/Pipfile:10]: if these are depejndencies of stringsifter they should be added when calling lock. Can we rely on stringsifter and allow stringsifter to handle its own deps?

File: docker/stringsifter/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply docker/stringsifter/Dockerfile:1]: can we check what it fails on?

File: docker/stringsifter/build.conf [hunk 4]
```diff
@@ -0,0 +1 @@
+version=1.0.0
```
Comment (JudahSchwartz) [inline docker/stringsifter/build.conf:1]: would reccomend using dynamic versioning to lock onto stringsifter's version. See how its done in demisto-sdk image

File: docker/stringsifter/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+COPY requirements.txt .
+
+RUN apt-get update && apt-get upgrade -y && apt-get install -y --no-install-recommends python3-dev make gcc g++ wget git libgomp1\
+  && pip install --no-cache-dir -r requirements.txt \
+  && apt-get purge -y --auto-remove gcc g++ make python3-dev
```
Comment (JudahSchwartz) [inline docker/stringsifter/Dockerfile:7]: do we still need wget and git? can they be removed

File: .github/dependabot.yml [hunk 6]
```diff
@@ -707,3 +707,9 @@ updates:
     directory: /docker/xpanse-ml-ev2
     schedule:
      interval: daily
+  - package-ecosystem: pip
+    directory: /docker/stringsifter
+    schedule:
+      interval: daily
+    ignore:
+      - dependency-name: "lightgbm"
```
Comment (JudahSchwartz) [inline .github/dependabot.yml:715]: why ignore?

File: docker/stringsifter/Dockerfile (hunk 5 above)
Comment (dansterenson) [reply docker/stringsifter/Dockerfile]: removed

File: .github/dependabot.yml (hunk 6 above)
Comment (dansterenson) [reply .github/dependabot.yml:719]: @wissamg did it, not sure why, what exactly does it do?

Comment (JudahSchwartz) [reply .github/dependabot.yml:719]: will not allow dependabot to update this subdependency

**Review by JudahSchwartz — APPROVED** [review]
looks good to me!

**Pull request merged**